diff --git a/applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/sqm.js b/applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/sqm.js
new file mode 100644
index 000000000..bde179602
--- /dev/null
+++ b/applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/sqm.js
@@ -0,0 +1,28 @@
+/* Licensed to the public under the Apache License 2.0. */
+
+'use strict';
+'require baseclass';
+
+return baseclass.extend({
+	title: _('SQM'),
+
+	rrdargs: function(graph, host, plugin, plugin_instance, dtype) {
+		var overview = {
+			per_instance: false,
+			title: "%H: SQM qdisc %pi Overview",
+			rrdopts: [ "--logarithmic" ],
+			vlabel: " ",
+			alt_autoscale: true,
+			number_format: "%5.0lf",
+			data: {
+				types: [ "qdisc_bytes", "qdisc_backlog", "qdisc_drops" ],
+				options: {
+					qdisc_bytes: { title: "kb/s:", overlay: true, noarea: false, color: "0000ff", transform_rpn: "125,/" },
+					qdisc_backlog: { title: "Backlog/B:", overlay: true, noarea: true, color: "8000ff" },
+					qdisc_drops: { title: "Drops/s:", overlay: true, noarea: true, color: "00ffff", }
+				}
+			}
+		};
+		return [ overview ];
+	}
+});
diff --git a/applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/sqmcake.js b/applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/sqmcake.js
new file mode 100644
index 000000000..fe433352b
--- /dev/null
+++ b/applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/sqmcake.js
@@ -0,0 +1,78 @@
+/* Licensed to the public under the Apache License 2.0. */
+
+'use strict';
+'require baseclass';
+
+return baseclass.extend({
+	title: _('SQM-Cake'),
+
+	rrdargs: function(graph, host, plugin, plugin_instance, dtype) {
+		var tindrops = {
+			per_instance: true,
+			title: "%H: CAKE %pi %di Drops v Backlog",
+			vlabel: "Bytes v Drops",
+			rrdopts: [ "--logarithmic" ],
+			number_format: "%5.0lf",
+			data: {
+				types: [ "qdisct_backlog", "qdisct_drops" ],
+				sources: { qdisct_drops: [ "drops", "ecn", "ack" ] },
+				options: {
+					qdisct_backlog: { title: "Backlog:", overlay: true, noarea: true, color: "0000ff" },
+					qdisct_drops__drops: { title: "Drops:", overlay: true, noarea: true, color: "00ffff" },
+					qdisct_drops__ecn: { title: "Ecn:", overlay: true, noarea: true, color: "00ff00" },
+					qdisct_drops__ack: { title: "Ack:", overlay: true, noarea: true, color: "ff00ff" }
+				}
+			}
+		};
+
+		var tinlatency = {
+			per_instance: true,
+			title: "%H: CAKE %pi %di Latency",
+			vlabel: "ms",
+			number_format: "%4.3lf",
+			data: {
+				types: [ "qdisct_latencyus" ],
+				sources: { qdisct_latencyus: [ "tg", "pk", "av", "sp" ] },
+				options: {
+					qdisct_latencyus__tg: { title: "Target:", overlay: true, noarea: true, color: "000000", transform_rpn: "1000,/" },
+					qdisct_latencyus__pk: { title: "Peak:", overlay: true, noarea: true, color: "ff0000", transform_rpn: "1000,/" },
+					qdisct_latencyus__av: { title: "Avg:", overlay: true, noarea: true, color: "00ff00", transform_rpn: "1000,/" },
+					qdisct_latencyus__sp: { title: "Sparse:", overlay: true, noarea: true, color: "0000ff", transform_rpn: "1000,/" }
+				}
+			}
+		};
+
+		var tinflows = {
+			per_instance: true,
+			title: "%H: CAKE %pi %di Flow Counts",
+			vlabel: "Flows",
+			number_format: "%4.0lf",
+			data: {
+				types: [ "qdisct_flows" ],
+				sources: { qdisct_flows: [ "sp", "bu", "un" ] },
+				options: {
+					qdisct_flows__sp: { title: "Sparse:", overlay: true, noarea: true, color: "00ff00" },
+					qdisct_flows__bu: { title: "Bulk:", overlay: true, noarea: true, color: "0000ff" },
+					qdisct_flows__un: { title: "Unresponsive:", overlay: true, noarea: true, color: "ff0000" }
+				}
+			}
+		};
+
+		var tinbytes = {
+			per_instance: true,
+			title: "%H: CAKE %pi %di Traffic",
+			vlabel: "kb/s",
+			number_format: "%5.0lf",
+			rrdopts: [ "--logarithmic" ],
+			data: {
+				types: [ "qdisct_bytes", "qdisct_thres" ],
+				options: {
+					qdisct_bytes: { title: "kb/s:", noarea: false, color: "0000ff", transform_rpn: "125,/" },
+					qdisct_thres: { title: "Thres:", overlay: true, noarea: true, color: "000000", transform_rpn: "125,/" }
+				}
+			}
+		};
+
+		return [ tinbytes, tinlatency, tindrops, tinflows ];
+	}
+});
diff --git a/applications/luci-app-statistics/root/etc/config/luci_statistics b/applications/luci-app-statistics/root/etc/config/luci_statistics
index dc515e35c..7747ea3b0 100644
--- a/applications/luci-app-statistics/root/etc/config/luci_statistics
+++ b/applications/luci-app-statistics/root/etc/config/luci_statistics
@@ -55,7 +55,7 @@ config statistics 'collectd_apcups'
 	option Port '3551'
 
 config statistics 'collectd_conntrack'
-	option enable '0'
+	option enable '1'
 
 config statistics 'collectd_contextswitch'
 	option enable '0'
@@ -64,7 +64,8 @@ config statistics 'collectd_cpu'
 	option enable '1'
 
 config statistics 'collectd_cpufreq'
-	option enable '0'
+	option enable '1'
+	option ExtraItems '1'
 
 config statistics 'collectd_curl'
 	option enable '0'
@@ -87,11 +88,17 @@ config statistics 'collectd_dns'
 	option IgnoreSources '127.0.0.1'
 
 config statistics 'collectd_entropy'
-	option enable '0'
+	option enable '1'
 
 config statistics 'collectd_exec'
 	option enable '0'
 
+#example for sqm_collect plugin
+#config collectd_exec_input
+#        option cmduser 'nobody'
+#        option cmdgroup 'nogroup'
+#        option cmdline '/usr/bin/collectd/sqm_collectd.sh eth0 ifb4eth0'
+
 config statistics 'collectd_interface'
 	option enable '1'
 	option Interfaces 'br-lan'
@@ -168,9 +175,9 @@ config statistics 'collectd_tcpconns'
 	option LocalPorts '22 80'
 
 config statistics 'collectd_thermal'
-	option enable '0'
-	option IgnoreSelected '0'
-	option Device ''
+	option enable '1'
+	option IgnoreSelected '1'
+	option Device 'cooling_device0 cooling_device1'
 
 config statistics 'collectd_uptime'
 	option enable '0'
diff --git a/modules/luci-base/root/etc/config/luci b/modules/luci-base/root/etc/config/luci
index 46947ae4c..2bd9b85f5 100644
--- a/modules/luci-base/root/etc/config/luci
+++ b/modules/luci-base/root/etc/config/luci
@@ -23,6 +23,7 @@ config internal ccache
 	option enable 1
 		
 config internal themes
+	option Bootstrap '/luci-static/bootstrap'
 
 config internal apply
 	option rollback 90
diff --git a/themes/luci-theme-bootstrap/root/etc/uci-defaults/30_luci-theme-bootstrap b/themes/luci-theme-bootstrap/root/etc/uci-defaults/30_luci-theme-bootstrap
index 4381a15c2..45784f68a 100755
--- a/themes/luci-theme-bootstrap/root/etc/uci-defaults/30_luci-theme-bootstrap
+++ b/themes/luci-theme-bootstrap/root/etc/uci-defaults/30_luci-theme-bootstrap
@@ -4,7 +4,7 @@ if [ "$PKG_UPGRADE" != 1 ]; then
 	uci get luci.themes.Bootstrap >/dev/null 2>&1 || \
 	uci batch <<-EOF
 		set luci.themes.Bootstrap=/luci-static/bootstrap
-		set luci.main.mediaurlbase=/luci-static/bootstrap
+#		set luci.main.mediaurlbase=/luci-static/bootstrap
 		commit luci
 	EOF
 fi
diff --git a/themes/luci-theme-material/root/etc/uci-defaults/30_luci-theme-material b/themes/luci-theme-material/root/etc/uci-defaults/30_luci-theme-material
index 7f07239ec..e7c376df8 100755
--- a/themes/luci-theme-material/root/etc/uci-defaults/30_luci-theme-material
+++ b/themes/luci-theme-material/root/etc/uci-defaults/30_luci-theme-material
@@ -4,7 +4,7 @@ if [ "$PKG_UPGRADE" != 1 ]; then
 	uci get luci.themes.Material >/dev/null 2>&1 || \
 	uci batch <<-EOF
 		set luci.themes.Material=/luci-static/material
-		set luci.main.mediaurlbase=/luci-static/material
+#		set luci.main.mediaurlbase=/luci-static/material
 		commit luci
 	EOF
 fi
